From 69c78ae86d3e08c9d53031ed378d387f0e94ab7b Mon Sep 17 00:00:00 2001
From: Omair Majid <omajid@redhat.com>
Date: Fri, 20 Nov 2020 15:00:09 -0500
Subject: [PATCH] Remove AppHosts from tools

The tool projects are building the apphost, but we always invoke the
tool through `dotnet tool.dll` instead of the apphost (`tool`). Removing
the apphost should have no effts.

Keeping the AppHosts introduces additional prebuilts for source build
and it would be nice to not have to maintain this patch in source-build.
See https://github.com/dotnet/source-build/issues/1905 for more details.
---
 src/Tools/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj   | 1 +
 .../GenerateDocumentationAndConfigFiles.csproj                   | 1 +
 .../GenerateGlobalAnalyzerConfigs.csproj                         | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/Tools/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj b/src/Tools/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
index 42b47952b..ba0a9612d 100644
--- a/src/Tools/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
+++ b/src/Tools/GenerateAnalyzerNuspec/GenerateAnalyzerNuspec.csproj
@@ -3,5 +3,6 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <NonShipping>true</NonShipping>
+    <UseAppHost>false</UseAppHost>
   </PropertyGroup>
 </Project>
diff --git a/src/Tools/GenerateDocumentationAndConfigFiles/GenerateDocumentationAndConfigFiles.csproj b/src/Tools/GenerateDocumentationAndConfigFiles/GenerateDocumentationAndConfigFiles.csproj
index ab935cd15..0056792ae 100644
--- a/src/Tools/GenerateDocumentationAndConfigFiles/GenerateDocumentationAndConfigFiles.csproj
+++ b/src/Tools/GenerateDocumentationAndConfigFiles/GenerateDocumentationAndConfigFiles.csproj
@@ -3,6 +3,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <NonShipping>true</NonShipping>
+    <UseAppHost>false</UseAppHost>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <MicrosoftCodeAnalysisVersion>$(MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion)</MicrosoftCodeAnalysisVersion>
   </PropertyGroup>
diff --git a/src/Tools/GenerateGlobalAnalyzerConfigs/GenerateGlobalAnalyzerConfigs.csproj b/src/Tools/GenerateGlobalAnalyzerConfigs/GenerateGlobalAnalyzerConfigs.csproj
index 9793af20a..c76df9dfd 100644
--- a/src/Tools/GenerateGlobalAnalyzerConfigs/GenerateGlobalAnalyzerConfigs.csproj
+++ b/src/Tools/GenerateGlobalAnalyzerConfigs/GenerateGlobalAnalyzerConfigs.csproj
@@ -4,6 +4,7 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
     <NonShipping>true</NonShipping>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
+    <UseAppHost>false</UseAppHost>
     <MicrosoftCodeAnalysisVersion>$(MicrosoftCodeAnalysisForRoslynDiagnosticsAnalyzersVersion)</MicrosoftCodeAnalysisVersion>
   </PropertyGroup>  
   <ItemGroup>
-- 
2.28.0

